ls1intum/Artemis
# `Exam mode`: Fix redirect to the exams in course-management for tutors and editors

### Checklist
#### General



#### Client

### Motivation and Context


This change fixes redirect to the exam in the course-management for tutors and editors: https://github.com/ls1intum/Artemis/issues/7562

### Description

The redirect didn't work, because both tutors and editors don't have access to the exam details page. They can only see the general exam management page. I added a check for the instructor access rights. If a user is an instructor, they are redirected to the exam details page. If a user is a tutor or an editor, they are redirected to the general exam management page.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Editor
- 1 Tutor
- 1 Course with an exam where all above mentioned users are added

1. Go to Course (not Management)
2. Click on the Exam
3. There is a button to navigate to exam management
4. Click the button
5. Check whether you are redirected to the right page
6. Repeat for all above mentioned user types


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
